src/player.cpp

```diff
@@ -2485,12 +2485,6 @@ hint_rating player::rate_action_wear( const item &it ) const
 
 bool player::can_reload( const item &it, const itype_id &ammo ) const
 {
-    if( ammo.is_empty() ) {
```

Comment:
this could cause crashes because `ammo` is often passed through as `""`, and is_reloadable_with will crash if the input is empty.

Reply:
`is_reloadable_helper` has checks for ammo string (non)emptiness.

Reply:
On second thought - looks like not all cases are covered there, so it probably caused #40830.

Reply:
Possible fix:

```CPP
bool item::is_reloadable_helper( const itype_id &ammo, bool now ) const
{
    if( !is_reloadable() ) {
        return false;
    }

    // empty ammo is passed for listing possible ammo apparently, so it needs to return true.
    if( ammo.is_empty() ) {
        return true;
    }

    if( magazine_integral() ) {
        if( ammo_data() ) {
            if( ammo_current() != ammo ) {
                return false;
            }
        } else {
            const itype *at = find_type( ammo );
            if( ( !at->ammo || !ammo_types().count( at->ammo->type ) ) &&
                !magazine_compatible().count( ammo ) ) {
                return false;
            }
        }
        return now ? ammo_remaining() < ammo_capacity( find_type( ammo )->ammo->type ) : true;
    }
    return magazine_compatible().count( ammo );
}
```